confgenerator/logging_processors.go

```diff
@@ -52,13 +60,109 @@ func (r LoggingProcessorParseRegex) Type() string {
 }
 
 func (p LoggingProcessorParseRegex) Components(tag, uid string) []fluentbit.Component {
-	filter, parser := p.LoggingProcessorParseShared.Components(tag, uid)
+	parser, parserName := p.ParserShared.Component(tag, uid)
 	parser.Config["Format"] = "regex"
 	parser.Config["Regex"] = p.Regex
+
 	return []fluentbit.Component{
-		filter,
 		parser,
+		fluentbit.ParserFilterComponent(tag, p.Field, []string{parserName}),
+	}
+}
+
+type RegexParser struct {
+	Regex  string
+	Parser fluentbit.ParserShared
+}
+
+// A LoggingProcessorParseRegexComplex applies a set of regexes to the specified field, storing the named capture groups as keys in the log record.
+type LoggingProcessorParseRegexComplex struct {
+	ConfigComponent
```

Comment:
I don't think `ConfigComponent` is used anymore.

All of these new types and methods would be candidates to move into the `fluentbit` package since they're no longer user-facing. Up to you, I think based on how likely it is that we'll want to expose these to user config. (Personally, I'd guess not likely, because they are incredibly fluentbit-specific.)

Reply:
I would probably prefer it here, and also refactoring it out without making a circular dependency looks complex, so I think we'll keep it there for now and possibly tackle it in a refactor PR if it makes sense in the future.

Reply:
Sounds good